src/order/rel_classes.lean

```diff
@@ -74,6 +74,30 @@ protected theorem is_asymm.is_irrefl [is_asymm α r] : is_irrefl α r :=
 protected theorem is_total.is_trichotomous (r) [is_total α r] : is_trichotomous α r :=
 ⟨λ a b, or.left_comm.1 (or.inr $ total_of r a b)⟩
 
+theorem is_refl_swap {α : Type u} (r : α → α → Prop) [is_refl α r] :
+  is_refl α (λ x y, r y x) :=
+⟨@refl α r _⟩
+
+theorem is_irrefl_swap {α : Type u} (r : α → α → Prop) [is_irrefl α r] :
+  is_irrefl α (λ x y, r y x) :=
+⟨@irrefl α r _⟩
```

Comment:
```suggestion
/-- Note this cannot be an instance as it would loop with itself -/
theorem is_refl_swap {α : Type u} (r : α → α → Prop) [is_refl α r] :
  is_refl α (λ x y, r y x) :=
⟨@refl α r _⟩

/-- Note this cannot be an instance as it would loop with itself -/
theorem is_irrefl_swap {α : Type u} (r : α → α → Prop) [is_irrefl α r] :
  is_irrefl α (λ x y, r y x) :=
⟨@irrefl α r _⟩
```